src/main/java/io/github/gitbucket/markedj/Lexer.java

```diff
@@ -170,6 +171,18 @@ protected void token(String src, boolean top, boolean bq, LexerContext context){
                 }
             }
 
+			{
+				boolean doContinue = false;
+				for (Extension extension : options.getExtensions()) {
+					Extension.LexResult result = extension.lex(src, context, this::token);
+					src = result.getSource();
+					doContinue = doContinue || result.matches();
```

Comment:
Thanks. I thought just comparing the returned `src` with the original one would be okay but I like this interface as it's more clear.

By the way, I also think it's okay to continue immediately if `result.matches()` is true. Are there cases that multiple extensions are applied at the same time?

Reply:
You're absolutely right, there should be no valid cases to check more extensions after the first match.